include/zephyr/kernel_structs.h

```diff
@@ -80,6 +80,19 @@ extern "C" {
 
 /* end - states */
 
+/* pending reason: Which call making thread pending */
+
+/* z_impl_k_msgq_prepend make thread pending */
+#define _MSG_PREPEND (BIT(0))
+
+/* z_impl_k_msgq_put make thread pending */
+#define _MSG_PUT (BIT(1))
+
+/* used to clear MSG_QUEUE pending reason */
+#define _MSG_QUEUE (_MSG_PREPEND & _MSG_PREPEND)
```

Comment:
This doesn't look right - should this be a bitwise OR of the two constants above?

Reply:
Fix. Maybe it will remove. Because I find we don't need to bit to know the pending reason.